### PR TITLE
fix(route): 微博自定义分组(使用Cookie)

### LINKED ISSUE #12181: 微博“自定义分组”时间线
[Issuehunt badges](https://oss.issuehunt.io/r/DIYgod/RSSHub/issues/12181)

### 类型

社交媒体

### 网站地址

www.weibo.com

### 网站描述

微博

### 需要生成什么内容？

微博“自定义分组”时间线：抓取各个自定义分组的时间线。

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Submitted pull Requests
- [#12514 fix(route): 微博自定义分组(使用Cookie)](https://oss.issuehunt.io/r/DIYgod/RSSHub/pull/12514)
---

### Tips

- Checkout the [Issuehunt explorer](https://oss.issuehunt.io/r/DIYgod/RSSHub/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://oss.issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by monologconnor**
## 该 PR 相关 Issue / Involved Issue

Close #12181

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/weibo/group/4541216424989965
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [x] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

此路由为路由`/weibo`下的一个新子路由, 参考`/weibo/user`路由的基本逻辑编写, 用于获取用户自定义分组下的微博时间线.
由于微博官方并未提供相关api, 只能通过使用`用户cookie`自行部署实现, 用户cookie获取以及配置流程也在对应文档中写明(目前只填写了中文文档). 
头一次编写JavaScript的项目, 如果代码内容有不规范还请见谅. 因个人不太懂网页微博的cookie过期时间以及cookie的更新流程, 该方案在一次部署下能维持多久有效暂无法保证.

第一个`commit`更改了`user.js`, 使订阅的条目链接可以用安卓第三方微博客户端`Share`作为外部浏览器打开, 方便部分用户的使用

由于`docs`目录下`yarn docs:dev`的运行报错(为未声明对`markdown-it-emoji`的依赖而无法运行), 为确保项目稳定就不对`packages.json`进行修改, 暂且不确定文档可以正常编译.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#12181: 微博“自定义分组”时间线](https://oss.issuehunt.io/repos/127769231/issues/12181)
- [#12181: 微博“自定义分组”时间线](https://oss.issuehunt.io/repos/127769231/issues/12181)
---
</details>

### CODE REVIEW

File: lib/v2/weibo/group.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            if (displayVideo === '1') {
+                if (item.retweeted_status) {
+                    description = weiboUtils.formatVideo(description, item.retweeted_status);
+                } else {
+                    description = weiboUtils.formatVideo(description, item);
+                }
+            }
+
+            if (displayComments === '1') {
+                description = await weiboUtils.formatComments(ctx, descripion, item);
```
[the remark this answers is not in the document]
Comment (monologconnor) [reply lib/v2/weibo/group.js:68]: no idea why, `description` has been assigned above

File: lib/v2/weibo/group.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                pubDate,
+                author: item.user.screen_name
+            };
+        })
+    );
+
+    ctx.state.data = weiboUtils.sinaimgTvax({
+        title: `${group_name}`,
+        link: `https://weibo.com/mygroups?gid=${gid}`,
+        description: `微博自定义分组`,
```
Comment (TonyRL) [inline lib/v2/weibo/group.js:92]:
```suggestion
        description: '微博自定义分组',
```

File: lib/v2/weibo/maintainer.js [hunk 3]
```diff
@@ -5,4 +5,5 @@ module.exports = {
     '/super_index/:id/:type?/:routeParams?': ['zengxs'],
     '/timeline/:uid/:feature?/:routeParams?': ['zytomorrow', 'DIYgod ', 'Rongronggg9'],
     '/user/:uid/:routeParams?': ['DIYgod', 'iplusx', 'Rongronggg9'],
+    '/group/:gid/:gname?/:routeParams?': ['monologconnor']
```
Comment (TonyRL) [inline lib/v2/weibo/maintainer.js:8]: Please insert new entries in alphabetical order.

File: lib/v2/weibo/router.js [hunk 4]
```diff
@@ -5,4 +5,5 @@ module.exports = (router) => {
     router.get('/super_index/:id/:type?/:routeParams?', require('./super_index'));
     router.get('/timeline/:uid/:feature?/:routeParams?', require('./timeline'));
     router.get('/user/:uid/:routeParams?', require('./user'));
+    router.get('/group/:gid/:gname?/:routeParams?', require('./group'));
```
Comment (TonyRL) [inline lib/v2/weibo/router.js:8]: Please insert new entries in alphabetical order.

File: docs/social-media.md [hunk 5]
```diff
@@ -1572,6 +1572,20 @@ rule
 
 :::
 
+### 自定义分组
```
[the remark this answers is not in the document]
Comment (monologconnor) [reply docs/social-media.md:1577]: Damn, I was dumb, bro. Thanks!

All above fixed and committed.

File: lib/v2/weibo/group.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
+const querystring = require('querystring');
+const got = require('@/utils/got');
+const config = require('@/config').value;
+const weiboUtils = require('./utils');
+const { fallback, queryToBoolean } = require('@/utils/readable-social');
+
+module.exports = async (ctx) => {
```
Comment (TonyRL) [inline lib/v2/weibo/group.js:7]: https://github.com/DIYgod/RSSHub/blob/9bdba060c7962d734745c8d6b9264da9a532b51d/lib/routes/disqus/posts.js#L5-L7
Similarly, you can throw an error if the cookie is not set

**Comment by Rongronggg9** [comment]
> 使订阅的条目链接可以用安卓第三方微博客户端 `Share` 作为外部浏览器打开, 方便部分用户的使用

You have introduced a breaking change making all existing subscribers re-receive all entries again as you effectively changed guids of all feed entries.
`m.weibo.cn` won't redirect to `weibo.com` on PC, which annoys PC users.

**Comment by Rongronggg9** [comment]
@TonyRL I suggest any unnecessary changes on guid or link should be discussed among other maintainers before it can be merged. Besides, any insignificant commits irrelevant to the patchset title should be refused.

**Comment by monologconnor** [comment]
> > 使订阅的条目链接可以用安卓第三方微博客户端 `Share` 作为外部浏览器打开, 方便部分用户的使用
> 
> You have introduced a breaking change making all existing subscribers receive all entries again as you effectively changed guids of all feed entries. `m.weibo.cn` won't redirect to `weibo.com` on PC, which annoys PC users.

I got ur mean. At very first, the commit changing `weibo.com` to `m.weibo.cn` is only for my personal usage, but I didn't notice that PR will take all my commits due to unexperienced.

To deal this, I probably will make another PR soon, which takes another optional parameter for all `/weibo` routes which determines either `weibo.com` or `m.weibo.cn` should be returned, and `weibo.com` would be the default option if not specified.

Is that a possible solution to u? If there is no potential issue on that, I'll do the coding and documentation later today.

**Comment by Rongronggg9** [comment]
> To deal this, I probably will make another PR soon

I have a WIP patch to achieve that. I plan to submit it tonight.

**Pull request merged**